demos/browser/app/meetingV2/videofilter/SegmentationUtil.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { BrowserBehavior } from 'amazon-chime-sdk-js';
+
+const SEGMENTATION_DEPENDENCIES = [
+  {
+    src: 'https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@2.7.0/dist/tf.min.js',
+    integrity: 'sha384-uI1PW0SEa/QzAUuRQ6Bz5teBONsa9D0ZbVxMcM8mu4IjJ5msHyM7RRtZtL8LnSf3',
+    crossOrigin: 'anonymous',
+  },
+
+  {
+    src: 'https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-core@2.7.0/dist/tf-core.min.js',
+    integrity: 'sha384-DlI/SVdTGUBY5hi4h0p+nmC6V8i0FW5Nya/gYElz0L68HrSiXsBh+rqWcoZx3SXY',
+    crossOrigin: 'anonymous',
+  },
+
+  {
+    src:
+      'https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-backend-webgl@2.7.0/dist/tf-backend-webgl.min.js',
+    integrity: 'sha384-21TV9Kpzn8SF68G1U6nYN3qPZnb97F06JuW4v0FDDBzW+CUwv8GcKMR+BjnE7Vmm',
+    crossOrigin: 'anonymous',
+  },
+
+  {
+    src: 'https://cdn.jsdelivr.net/npm/@tensorflow-models/body-pix@2.0.5/dist/body-pix.min.js',
+    integrity: 'sha384-dPJ/sICXCqdh39bsuGLVFcOiRyeL/XcFrwiFrJq9oh7k1TCtsUKhX6UV2X4UuKU4',
+    crossOrigin: 'anonymous',
+  },
+];
+
+export async function loadBodyPixDependency(): Promise<void> {
+  // the tf library loading order must be followed
+  SEGMENTATION_DEPENDENCIES.forEach(async element => {
+    await new Promise<void>((resolve, _) => {
```

Comment:
Callbacks you pass to `forEach()` run simultaneously, and the `try/catch` statement in `populateVideoFilterInputList()` won't catch any error occurring from the callback. Also, `loadBodyPixDependency()` will immediately end without waiting for loading scripts.

What are you trying to achieve here?
1. Wait until all four scripts are loaded and then complete `loadBodyPixDependency()`.
2. Load each script at a time. In the end, complete `loadBodyPixDependency()`.

Reply:
I am trying to achieve 2, load each script at a time, and then complete  `loadBodyPixDependency`. 
Isn't `await` going to wait for the promise to finish before executing next element in the `SEGMENTATION_DEPENDENCIES` ?

Reply:
oh ... 
```
forEach does not wait for promises. Kindly make sure you are aware of the implications while using promises(or async functions) as forEach callback.
```

I will fix this. thanks

Reply:
For (2), you can just replace `forEach` with a normal for loop.

Try the following snippet in your browser console.
```
(async function () {
  console.log('start');
  [1,2,3].forEach(async i => {
    await new Promise((r, reject) => {
      console.log(`start ${i}`);
      setTimeout(() => {
        console.log(`end ${i}`);
        r();
      }, 1500);
    });
  });
  console.log('end');
})();
```

Expected results:
```
start
start 1
end 1
start 2
end 2
start 3
end 3
end
```

Actual results: 
```
start
start 1
start 2
start 3
end
// After delay
end 1
end 2
end 3
```